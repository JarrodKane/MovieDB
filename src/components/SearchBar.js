import React from "react";
//import { Popup } from "semantic-ui-react";

const SearchBar = props => {
  const {
    un,
    pw,
    search,
    submitLogin,
    onUNChange,
    onPWChange,
    onSearchChange,
    submitSearch,
    errorData,
    handleLogout,
    isSignedIn,
    searchFalse = true
  } = props;

  let errorMesg;
  if (errorData === true) {
    errorMesg = (<div>Erros</div>
      //TODO: CHANGE FROM SEMANTIC USE
      /*
      <Popup
        trigger={<div icon="add" />}
        open={true}
        content="Fill in username or password"
      />
      */
    );
  } else {
    errorMesg = "";
  }
  let logBtn;
  if (isSignedIn === false) {
    logBtn = (
      <button className="ui primary button" onClick={submitLogin}>
        Login
      </button>
    );
  } else {
    logBtn = (
      <button className="ui red button" onClick={handleLogout}>
        Logout
      </button>
    );
  }

  return (
    <div className="ui stackable menu">
      {searchFalse ? (
        <div className="right item ">
          <div className="ui icon  input">
            <input
              type="text"
              placeholder="Search..."
              value={search}
              onChange={onSearchChange}
            />
            <i aria-hidden="true" className="search icon"></i>
          </div>
          <button className="ui big button" onClick={submitSearch}>
            Search
          </button>
        </div>
      ) : (
          <div className="right item "></div>
        )}

      <div className="Username ui stackable menu">
        {errorMesg}
        <div className="right item">
          <div className="ui icon  input icon">
            <input
              type="text"
              placeholder="Username"
              id="un"
              name="un"
              value={un}
              onChange={onUNChange}
            />
            <i aria-hidden="true" className="user icon"></i>
          </div>
        </div>
        <div className="right item">
          <div className="ui action input ">
            <input
              type="password"
              placeholder="Password"
              id="pw"
              name="pw"
              value={pw}
              onChange={onPWChange}
            />
            {logBtn}
          </div>
        </div>
      </div>
    </div>
  );
};

export default SearchBar;
