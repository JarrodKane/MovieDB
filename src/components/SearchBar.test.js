import React from "react";
import { shallow } from "enzyme";
import SearchBar from "./SearchBar";
describe("Header", () => {
  it("should render correctly with no props", () => {
    const component = shallow(<SearchBar />);

    expect(component).toMatchSnapshot();
  });
});
