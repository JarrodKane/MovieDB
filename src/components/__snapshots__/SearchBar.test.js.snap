// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly with no props 1`] = `
<div
  className="ui stackable menu"
>
  <div
    className="right item "
  >
    <div
      className="ui icon  input"
    >
      <input
        placeholder="Search..."
        type="text"
      />
      <i
        aria-hidden="true"
        className="search icon"
      />
    </div>
    <button
      className="ui big button"
    >
      Search
    </button>
  </div>
  <div
    className="Username ui stackable menu"
  >
    <div
      className="right item"
    >
      <div
        className="ui icon  input icon"
      >
        <input
          id="un"
          name="un"
          placeholder="Username"
          type="text"
        />
        <i
          aria-hidden="true"
          className="user icon"
        />
      </div>
    </div>
    <div
      className="right item"
    >
      <div
        className="ui action input "
      >
        <input
          id="pw"
          name="pw"
          placeholder="Password"
          type="password"
        />
        <button
          className="ui red button"
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;
